k8scrdclient/k8scrdclient.go

```diff
@@ -154,6 +155,27 @@ func (c *CRDClient) ensureUpdated(ctx context.Context, desired *apiextensionsv1b
 func (c *CRDClient) validateStatus(crd *apiextensionsv1beta1.CustomResourceDefinition, b backoff.Interface) error {
 	var err error
 
+	version, err := c.k8sExtClient.Discovery().ServerVersion()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	major, err := strconv.Atoi(version.Major)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	minor, err := strconv.Atoi(version.Minor)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if major < 2 && minor < 16 {
```

Comment:
if it's lower than 1.16, it will get caught from here.

Reply:
Let me get this straight. The error is because the CRD is invalid, and instead of fixing the CRD you want to not validate it instead? I think we should fix the CRD. The check here is ok, what is broken is the schema validation which we should fix.

Reply:
I think you mean by https://github.com/giantswarm/apiextensions/pull/364/files?